src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -126,7 +141,10 @@ def create_resource(self, name, **kwargs):
     def remove_resource(self, name, **kwargs):
         if not self.skip_delete and not self.dev_setting_name:
             group = self._get_resource_group(**kwargs)
-            self.live_only_execute(self.cli_ctx, 'az storage account delete -n {} -g {} --yes'.format(name, group))
+            cmd = 'az storage account delete -n {} -g {} --yes'.format(name, group)
+            if not self.subscription:
+                cmd += ' --subscription ' + self.subscription
```

Comment:
How about extract this logic into a common one?
Such as: `cmd = self.add_subscription(cmd)`

```python
def add_subscription(self, cmd):
   if not self.subscription:
      return cmd + '  --subscription ' + self.subscription
   return cmd
```

And this function can be placed in base class `NoTrafficRecordingPreparer`

Reply:
Good idea.